### ISSUE TITLE
[BUG] password reset: mail server error discloses user email

### ISSUE THREAD

**Issue by thielj** [lldap/lldap#887]
**Describe the bug**
An error while sending the password reset email discloses the email address associated with a user name.

![Screenshot 2024-03-29 123448](https://github.com/lldap/lldap/assets/3942997/be25caa9-e034-49f9-947b-325c3acee770)

**To Reproduce**
Setup the mail server to reject the email and start the password reset sequence.

**Expected behavior**
Sensitive user data should not be exposed.

**Comment by sploders101** [comment]
Hello! I just started using this project, and am fairly fluent in Rust. I would like to take this one on, but I'm curious what you see as an acceptable solution. Should we redact the email specifically from the error message or hide it entirely, instead logging it to a file? My vote would be for hiding the error message entirely. I'd rather have errors explicitly whitelisted before being logged to the client, as I'm sure others may disclose sensitive information as well.

**Comment by nitnelave** [comment]
Hey! Welcome to the project! Feel free to join the discord and ask for help there if you need. 

For this issue, I think it makes sense to "sanitize" the error sent to the frontend (e.g. "error sending the confirmation email"), and log the full error on the server.

**Comment by nitnelave** [comment]
Yeah, I guess it makes sense. It's not ideal to have a failure without feedback, but given that it's an unauthorized entry point I think it makes sense. 

While we're at it, we should also include a sleep(3s) before returning an error, otherwise the timing is too suspect.
